### PR TITLE
Breaks functional tests into smaller tests to conserve resources

### PR DESCRIPTION
**Pull request by gp510**
Functional tests have been timing out on CircleCI. By breaking up the tests they run more reliably.

### CODE REVIEW

File: .circleci/functional_tests_config.sh [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+sudo pip install --upgrade pip
+sudo -H pip install virtualenv
+virtualenv --python=python3.5 venv
+source venv/bin/activate
+pip install -r requirements.txt
+deactivate
+
+#Run functional tests from within virtualenv
+sudo venv/bin/python -m pytest --verbose --cache-clear \
+   --splunkd-url https://$SPLUNK_HEC_HOST:8089 \
```
Comment (luckyj5) [inline .circleci/functional_tests_config.sh]: Can we just iterate through the list (different tests) in a loop? Using `For loop` for eg

File: .circleci/functional_tests_malformed.sh [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+sudo pip install --upgrade pip
+sudo -H pip install virtualenv
+virtualenv --python=python3.5 venv
+source venv/bin/activate
+pip install -r requirements.txt
+deactivate
+
+#Run functional tests from within virtualenv
+sudo venv/bin/python -m pytest --verbose --cache-clear \
```
Comment (luckyj5) [inline .circleci/functional_tests_malformed.sh]: Can we just iterate through the list (different tests) in a loop? Using For loop for eg

**Review by luckyj5 — CHANGES_REQUESTED** [review]

File: .circleci/functional_tests_config.sh (hunk 1 above)
Comment (gp510) [reply .circleci/functional_tests_config.sh]: done

File: .circleci/functional_tests_malformed.sh (hunk 2 above)
Comment (gp510) [reply .circleci/functional_tests_malformed.sh]: done

**Review by luckyj5 — APPROVED** [review]

**Pull request merged**